### PR TITLE
Add flake8 no type

### LINKED ISSUE #1196: Set up linting rule to ban Python typing annotations
## Problem

- Some devs may end up submitting PRs which use Python type annotations _(points finger at nose)_
- Pending #1008, (with precedent already established in #913) we likely will not want to allow typing in Python code.

## Solution

- Instal [flake8-no-types](https://pypi.org/project/flake8-no-types/) which flags all forms of typing as a linting error.

### PR DESCRIPTION
**Pull request by ratika-12**
Fixes #1196 

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `master` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

### CODE REVIEW

**Comment by ratika-12** [comment]
@kgodey Please review

**Review by mathemancer — CHANGES_REQUESTED** [review]
@ratika-12 Did you verify that `flake8-no-types` is actually running on the github action? I don't think it is. You should look at the file `.github/workflows/run-flake8.yml` to start.

**Comment by ratika-12** [comment]
Thanks for reviewing. I will check it now.

File: .github/workflows/run-flake8.yml [hunk 1]
```diff
@@ -16,5 +16,6 @@ jobs:
         with:
           checkName: "flake8"
           path: "."
+          plugins: flake8-no-types
```
[the remark this answers is not in the document]
Comment (mathemancer) [reply .github/workflows/run-flake8.yml:19]: Indeed! See the result [here](https://github.com/centerofci/mathesar/runs/5661931430?check_suite_focus=true).

**Review by mathemancer — CHANGES_REQUESTED** [review]
Great work getting the new linting rules working.

Unfortunately, the new stricter linting has found some issues. See the result of [the action](https://github.com/centerofci/mathesar/runs/5661931430?check_suite_focus=true). Please resolve those issues, and then re-request review. You should just need to delete some type hints.

**Comment by ratika-12** [comment]
Okay!! Will check it.

File: .github/workflows/run-flake8.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (ratika-12) [reply .github/workflows/run-flake8.yml:19]: Thank you @mathemancer :)

**Comment by ratika-12** [comment]
> Great work getting the new linting rules working.
> 
> Unfortunately, the new stricter linting has found some issues. See the result of [the action](https://github.com/centerofci/mathesar/runs/5661931430?check_suite_focus=true). Please resolve those issues, and then re-request review. You should just need to delete some type hints.

Thanks for reviewing :) I have added the changes @mathemancer

File: .github/workflows/mathesar.code-workspace [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+    "folders": [
+        {
+            "path": "../.."
+        }
+    ]
+}
```
Comment (mathemancer) [inline .github/workflows/mathesar.code-workspace:7]: Why is this added?

**Review by mathemancer — CHANGES_REQUESTED** [review]
This is getting closer, thanks for that! 

The remaining tasks (as far as I can tell) are to remove the imports of types which were used in type hints but are now no longer needed.

Also, I don't understand why the code-workspace change is needed. What's going on there?

File: .github/workflows/mathesar.code-workspace (hunk 2 above)
Comment (ratika-12) [reply .github/workflows/mathesar.code-workspace:7]: oops sorry looks like my workspace got added by mistake. I will remove it

Comment (ratika-12) [reply .github/workflows/mathesar.code-workspace:7]: I will also remove the unused imports

**Comment by ratika-12** [comment]
@mathemancer I have removed the unwanted file and unused imports. Please review :)
